docs/guides/converting_and_upgrading/optax_update_guide.ipynb

```diff
@@ -0,0 +1,430 @@
+          "  grads = jax.grad(loss)(params, batch)\n",
+          "  updates, opt_state = tx.update(grads, opt_state)\n",
+          "  params = optax.apply_updates(params, updates)\n",
+          "  return params, opt_state\n",
+          "\n",
+          "tx = optax.sgd(learning_rate, momentum)\n",
+          "params = variables['params']\n",
+          "opt_state = tx.init(params)\n",
+          "\n",
+          "for batch in ds_train:\n",
+          "  params, opt_state = train_step(params, opt_state, batch)"
+        ]
+      },
+      {
+        "cell_type": "markdown",
+        "metadata": {
+          "id": "f48d6324bb90"
+        },
+        "source": [
+          "Composable gradient transformations\n",
+          "-----------------------------------\n",
+          "\n",
+          "The function |optax.sgd()|_ used in the code snippet above is simply a wrapper for the sequential application of two gradient transformations. Instead of using this alias, it is common to use |optax.chain()|_ to combine multiple of these generic building blocks."
+        ]
+      },
+      {
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {
+          "id": "f982f184943e"
+        },
+        "outputs": [],
+        "source": [
+          "# # Code with Optax (pre-defined alias)\n",
+          "#\n",
+          "# # Note that the aliases follow the convention to use positive\n",
+          "# # values for the learning rate by default.\n",
+          "# tx = optax.sgd(learning_rate, momentum)"
+        ]
+      },
+      {
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {
+          "id": "14d116acebe2"
+        },
+        "outputs": [],
+        "source": [
+          "# Code with Optax (combining transformations)\n",
+          "\n",
+          "tx = optax.chain(\n",
+          "    # 1. Step: keep a trace of past updates and add to gradients.\n",
+          "    optax.trace(decay=momentum),\n",
+          "    # 2. Step: multiply result from step 1 with negative learning rate.\n",
+          "    # Note that `optax.apply_updates()` simply adds the final updates to the\n",
+          "    # parameters, so we must make sure to flip the sign here for gradient\n",
+          "    # descent.\n",
+          "    optax.scale(-learning_rate),\n",
+          ")"
+        ]
+      },
+      {
+        "cell_type": "markdown",
+        "metadata": {
+          "id": "60bd02c4be2d"
+        },
+        "source": [
+          "## Weight decay\n",
+          "\n",
+          "- Some of `flax.optim` optimizers include the weight decay parameter.\n",
+          "- In Optax, some optimizers also have a weight decay parameter (such as `optax.adamw()`). For other optimizers that don't have it by default, the weight decay can be added as another \"gradient transformation\" `optax.add_decayed_weights()` that adds an update derived from the parameters."
```

Comment:
can we copy all hyperlinks from the doc page over, e.g. `optax.adamw()` should link to https://optax.readthedocs.io/en/latest/api.html#optax.adamw